src/website/app/demos/Dialog/examples/Dialog/title.js

```diff
@@ -0,0 +1,32 @@
+/* @flow */
+import { DialogTitle } from '../../../../../../library/Dialog';
+import DemoLayout from '../../components/DemoLayout';
+import Dialog from '../../components/DemoDialog';
+
+export default {
+  id: 'title',
+  title: 'Title',
+  description: `Dialog's \`title\` can be a simple string or you can pass a
+[DialogTitle](/components/dialog-title) to apply a different \`appearance\` or
+\`element\` or render non-string content. Note that Dialog automatically passes
+some props, such as \`id\` & \`variant\`, to DialogTitle.`,
```

Comment:
We may want to add a note to the first sentence explaining that if you do pass a simple string, a DialogTitle will automatically be created for you.